src/packages/server/auth/sso/oauth2-user-profile-callback.ts

```diff
@@ -59,7 +59,12 @@ export function addUserProfileCallback(opts: UserProfileCallbackOpts) {
       try {
         json = JSON.parse(body);
       } catch (ex) {
-        return done(new Error(`Failed to parse user profile -- ${body}`));
+        try { // OAuth1 body, need base64-like decoding
+            json = jwt_decode(body);
+        }
+        catch(ex){
```

Comment:
Here you catch the exception but seem to not refer to it, which is a typescript error (given our settings). E.g., something like this is better:

```ts
return done(new Error(`Failed to parse user profile -- ${body} -- error: ${ex}`));
```